src/daemon/daemon.cpp

```diff
@@ -88,6 +88,14 @@ bool Daemon::activate(const InterfaceConfig& config) {
       qWarning("Wireguard interface `%s` already exists.", WG_INTERFACE);
       return false;
     }
+    // add_if
+    if (!wgutils()->addInterface()) {
+      return false;
+    }
+    // set conf
+    if (!wgutils()->configureInterface(config)) {
```

Comment:
should we remove the interface if the configuration fails?

Reply:
YES! And the bash script does this with the "trap" and I forgot. Thanks.

Reply:
I'm no longer sure about this change. In the `if(m_connected)` block above, if something goes wrong we call deactivate. deactivate attempts to delete the interface. What about continuing to use that pattern?

Reply:
You're correct. We call `deactivate` on L77 (https://github.com/mozilla-mobile/mozilla-vpn-client/pull/633/files#diff-d93d28344d209ab944d30f1b5438025d11fdcf98a15cb00588e0c219f46d08fbR77) if connected and don't support server switching. I agree that's not the same as `activate` failing. But `activate` could try to call `deactivate` if it fails. I'm just thinking about where do we want clean-up logic to be and whether it's going to end up being duplicated.

I'm fine leaving this as it is for now and then we can review it later if we start to get a lot of similar looking code in activate and deactivate.